### ISSUE TITLE
Misleading error-message when mixing method overloads and named parameters

### ISSUE THREAD

**Issue by runehalfdan** [dotnet/roslyn#70515]
**Version Used**: 
.NET SDK 7.401

**Steps to Reproduce**:

When using named parameters to call a methods with multiple overload, the diagnostics gives a confusing error-message

```
using System;
public class C {
    public void M() {
        Dummy(1,2,3);
        
        Dummy(4, val2: 5, val3: 6);
        
        // error
        Dummy(4, val3: 5, val3: 6);
    }
    
    public static void Dummy(int val1, int val2){}
    
    public static void Dummy(int val1, int val2, int val3){}
    
    public static void Dummy(int val1, int val2, int val3, int val4){}
}
```
https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQMwAJboMLoN7LpGYZQAs6AsgBQCU+hxTAIgK4C27AntXADQw+qWgG5GTIuIltOPMn3QA3AIYAbGCHQBWBStWpNANlFSmp4jO7V5StQe267Rk0iYBfKVLSY4hzBUseAEsAOwAXW1V+dFCIvRhaPA9XYi9SX390QOpYyOjc+IUCu0TksxSibygM8iyOK2KoovDIwRiWvVRmuLUyUuQ3IA===

This issue was originally discovered in a big lump of code inside a HostBuilder.Configure, with lots of chaining and nested lambdas, making the actual issue hard to see. Specifically, this line triggered it (with repeated reloadOnChange: true)
`.AddJsonFile("appsettings.foo.json", reloadOnChange: true, reloadOnChange: true)`

**Diagnostic Id**:
CS1501

**Expected Behavior**:
"There is no argument given that corresponds to the required parameter 'val2' of 'C.Dummy(int, int, int)'"

**Actual Behavior**:
"No overload for method 'Dummy' takes 3 arguments."
